unlock-app/src/hooks/usePurchaseKey.ts

```diff
@@ -0,0 +1,21 @@
+import { useState, useContext } from 'react'
+import { WalletService } from '@unlock-protocol/unlock-js'
+import { RawLock } from '../unlockTypes'
+import { WalletServiceContext } from '../utils/withWalletService'
+
+export const usePurchaseKey = (lock: RawLock) => {
+  const [initiatedPurchase, setInitiatedPurchase] = useState(false)
+  const walletService: WalletService = useContext(WalletServiceContext)
+
+  const purchaseKey = () => {
+    setInitiatedPurchase(true)
+    walletService.purchaseKey({
+      lockAddress: lock.address,
+      owner: lock.owner!,
+      keyPrice: lock.keyPrice,
+      erc20Address: lock.currencyContractAddress,
+    })
+  }
+
+  return { purchaseKey, initiatedPurchase }
```

Comment:
We may want to track and return several different things here, like whether it completed the purchase or whether it errored in some way. Leaving them out for now until we have a concrete use for them.